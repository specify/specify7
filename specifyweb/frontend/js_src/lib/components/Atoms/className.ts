/**
 * If dialog contains a button with this className, it will use that icon
 * by default
 */
export const dialogIconTriggers = {
  // Icons are ordered by precedence
  none: '',
  error: 'dialog-icon-error',
  warning: 'dialog-icon-warning',
  success: 'dialog-icon-success',
  info: 'dialog-icon-info',
};

const buttonClassName = 'button';
const secondaryButton = `${buttonClassName} hover:brightness-90 dark:hover:brightness-125 bg-[color:var(--secondary-button-color)] text-gray-800 shadow-sm
dark:text-gray-100`;
const secondaryLightButton = `${buttonClassName} hover:brightness-90 dark:hover:brightness-125 bg-[color:var(--secondary-light-button-color)] text-gray-800 shadow-sm
dark:text-gray-100`;

const containerBaseUnstyled = `flex flex-col gap-4 overflow-scroll
  overflow-x-auto [overflow-y:overlay] [scrollbar-gutter:auto]`;
const containerBase = `${containerBaseUnstyled} bg-[color:var(--form-foreground)]
  rounded p-4 shadow-gray-400 shadow-2xl`;
const containerFull = 'flex flex-col gap-4 sm:h-full p-1 sm:p-4';
const formStyles =
  'text-[length:var(--form-font-size)] font-[family-name:var(--form-font-family)]';
const niceButton = `${buttonClassName} rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:dark:ring-neutral-500 disabled:ring-gray-400 disabled:text-gray-500 
    dark:disabled:!bg-neutral-700 gap-2 inline-flex items-center capitalize justify-center shadow-sm`;
const borderedGrayButton = `${secondaryButton} ring-1 ring-gray-400 shadow-sm dark:ring-0
    disabled:ring-gray-400 disabled:dark:ring-neutral-500`;

// REFACTOR: reduce this once everything is using React. Can move things into tailwind.config.js

export const className = {
  /*
   * Most fields in Specify are rendered on top of var(--form-background). For
   * some fields that are rendered on top of var(--background), this class
   * name is added to prevent background from clashing
   */
  hasAltBackground: 'has-alt-background',
  /*
   * Do not show validation errors until tried to submit the form or field lost
   * focus
   * These class names are negated so that if you forgot to add it in some
   * place, the validation errors do not get permanently silenced
   */
  notSubmittedForm: 'not-submitted',
  /*
   * Used to disable default aria-current="page" styles for links and
   * aria-pressed="true" for buttons
   */
  ariaHandled: 'aria-handled',
  notTouchedInput: 'not-touched',
  label: 'flex flex-col',
  labelForCheckbox: 'cursor-pointer inline-flex gap-1 items-center',
  textArea: 'max-w-full min-w-[theme(spacing.20)] min-h-[theme(spacing.8)]',
  button: buttonClassName,
  link: 'link',
  icon: 'icon link',
  secondaryButton,
  secondaryLightButton,
  niceButton,
  smallButton: `${niceButton} !py-1 !px-2`,
  borderedGrayButton,
  defaultSmallButtonVariant: `${borderedGrayButton}`,
  dangerButton: `${dialogIconTriggers.error} hover:brightness-90 dark:hover:brightness-150 bg-[color:var(--danger-button-color)] text-white`,
  infoButton: `${dialogIconTriggers.info} hover:brightness-90 dark:hover:brightness-150 bg-[color:var(--info-button-color)] text-white`,
  warningButton: `${dialogIconTriggers.warning} hover:brightness-90 dark:hover:brightness-150 bg-[color:var(--warning-button-color)] text-white`,
  successButton: `${dialogIconTriggers.success} hover:brightness-90 dark:hover:brightness-150 bg-[color:var(--success-button-color)] text-white`,
  saveButton: `hover:brightness-90 dark:hover:brightness-150 text-white bg-[color:var(--save-button-color)]`,
  fancyButton: `bg-gray-300 hover:bg-brand-200 dark:bg-neutral-600
    hover:dark:bg-brand:400 text-gray-800 dark:text-white text-center`,
  containerFull,
  containerFullGray: `${containerFull} bg-[color:var(--form-background)]`,
  containerBaseUnstyled,
  containerBase,
  containerCenter: `${containerBase} max-w-[var(--form-max-width)]
    mx-auto w-full ${formStyles}`,
  formHeader:
    'border-b-2 border-brand-300 flex items-center pb-2 gap-2 md:gap-4 md:justify-between flex-col md:flex-row',
  formTitle: 'text-lg font-bold flex items-center gap-2',
  formStyles,
  limitedWidth: `max-w-[var(--max-field-width)]`,
  headerPrimary: 'font-semibold text-black dark:text-white',
  headerGray: 'text-gray-500 dark:text-neutral-400',
  // These values must be synchronised with main.css
  dataEntryGrid: 'data-entry-grid',
  formFooter:
    'border-brand-300 border-t-2 flex print:hidden pt-2 gap-2 flex-wrap justify-center flex-col md:flex-row',
  dataEntryAdd: '!text-green-700 print:hidden',
  dataEntryView: '!text-cyan-400 print:hidden',
  dataEntryEdit: '!text-orange-400 print:hidden',
  dataEntryClone: '!text-amber-700 print:hidden',
  dataEntrySearch: '!text-blue-500 print:hidden',
  dataEntryRemove: '!text-red-700 print:hidden',
  dataEntryVisit: '!text-blue-700 print:hidden',
} as const;
