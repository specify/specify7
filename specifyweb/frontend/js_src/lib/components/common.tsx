import React from 'react';

import icons from '../icons';
import { spanNumber } from '../wbplanviewhelper';
import dataModelStorage from '../wbplanviewmodel';

const MAX_HUE = 360;
const getHue = spanNumber(
  'a'.charCodeAt(0) * 2,
  'z'.charCodeAt(0) * 2,
  0,
  MAX_HUE
);

export function TableIcon({
  tableName,
  tableLabel,
}: {
  readonly tableName: string;
  readonly tableLabel?: string | false;
}): JSX.Element {
  const tableIconSource = icons.getIcon(tableName);
  const resolvedLabel =
    tableLabel === false
      ? undefined
      : tableLabel ?? dataModelStorage.tables[tableName]?.label ?? '';
  if (tableIconSource !== '/images/unknown.png')
    return (
      <span
        className="table-icon table-icon-image"
        role="img"
        style={{ backgroundImage: `url('${tableIconSource}')` }}
        title={resolvedLabel}
        aria-label={resolvedLabel}
      />
    );

  const colorHue = getHue(tableName.charCodeAt(0) + tableName.charCodeAt(0));
  const color = `hsl(${colorHue}, 70%, 50%)`;
  return (
    <span
      style={{ backgroundColor: color }}
      role="img"
      className="table-icon table-icon-generated"
      title={resolvedLabel}
      aria-label={resolvedLabel}
    >
      {tableName.slice(0, 2).toUpperCase()}
    </span>
  );
}

export const TableIconUndefined = (
  <span className="table-icon table-icon-undefined">⃠</span>
);

export const TableIconSelected = (
  <span className="table-icon table-icon-selected">✓</span>
);

export const TableIconEmpty = <span className="table-icon table-icon-emtpy" />;

export function DateElement({
  date,
  fallback = undefined,
}: {
  readonly date: string | undefined;
  readonly fallback?: React.ReactNode;
}): JSX.Element {
  if (typeof date !== 'string' || Number.isNaN(Date.parse(date)))
    return <>{fallback}</>;
  const dateObject = new Date(date);
  return (
    <time
      dateTime={dateObject.toISOString()}
      title={dateObject.toLocaleString()}
      aria-label={dateObject.toLocaleString()}
    >
      {dateObject.toDateString()}
    </time>
  );
}
