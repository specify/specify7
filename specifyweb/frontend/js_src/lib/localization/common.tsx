import React from 'react';
import { createDictionary, createHeader, createJsxHeader } from './utils';

// Refer to "Guidelines for Programmers" in ./README.md before editing this file

const commonText = createDictionary({
  specifySeven: 'Specify&nbsp7',
  pageNotFound: 'Page Not Found',

  collectionAccessDeniedDialogTitle: 'Access denied',
  collectionAccessDeniedDialogHeader: createHeader(
    'You do not have access to this collection'
  ),
  collectionAccessDeniedDialogMessage: (collectionName: string) =>
    `The currently logged in account does not have access to the
    ${collectionName} collection.`,
  noAgentDialogTitle: 'No Agent assigned',
  noAgentDialogHeader: createHeader(
    'Current user does not have an agent assigned'
  ),
  noAgentDialogMessage:
    'Please log in as admin and assign an agent to this user',

  no: 'No',
  cancel: 'Cancel',
  create: 'Create',
  close: 'Close',
  apply: 'Apply',
  applyAll: 'Apply All',
  clearAll: 'Clear all',
  save: 'Save',
  add: 'Add',
  open: 'Open',
  delete: 'Delete',
  next: 'Next',
  previous: 'Previous',
  tools: 'Tools',
  loading: 'Loading...',
  tableName: 'Table Name',
  name: 'Name',
  created: 'Created',
  uploaded: 'Uploaded',
  createdBy: 'Created by:',
  modifiedBy: 'Modified by:',
  stop: 'Stop',
  remove: 'Remove',
  search: 'Search',
  noResults: 'No Results',
  new: 'New',
  reports: 'Reports',
  labels: 'Labels',
  edit: 'Edit',
  ignore: 'Ignore',
  logIn: 'Log In',
  start: 'Start',
  update: 'Update',
  generate: 'Generate',
  loadingInline: '(loading...)',
  listTruncated: '(list truncated)',
  metadataInline: 'Metadata:',
  metadata: 'Metadata',
  query: 'Query',

  // Toolbar
  notifications: (count: number) => `Notifications: ${count}`,
  attachments: 'Attachments',
  dataEntry: 'Data Entry',
  makeDwca: 'Make DwCA',
  definitionResourceNotFound: (resourceName: string) =>
    `Definition resource "${resourceName}" was not found.`,
  metadataResourceNotFound: (resourceName: string) =>
    `Metadata resource "${resourceName}" was not found.`,
  updateExportFeed: 'Update Feed Now',
  updateExportFeedDialogTitle: 'Export Feed',
  updateExportFeedDialogHeader: createHeader(
    'Update all export feed items now?'
  ),
  updateExportFeedDialogMessage: 'Update all export feed items now?',
  feedExportStartedDialogTitle: 'Export Feed',
  feedExportStartedDialogHeader: createHeader('Export feed update started'),
  feedExportStartedDialogMessage: `
    Update started. You will receive a notification for each feed item
    updated.`,
  dwcaExportStartedDialogTitle: 'DwCA',
  dwcaExportStartedDialogHeader: createHeader('DwCA export started'),
  dwcaExportStartedDialogMessage: `
    Export started. You will receive a notification
    when the export is complete.`,
  interactions: 'Interactions',
  generateMasterKey: 'Generate Master Key',
  generateMasterKeyDialogTitle: 'Master Key',
  generateMasterKeyDialogHeader: createHeader('Generate Master Key'),
  userPassword: 'User Password:',
  masterKeyDialogTitle: 'Master Key',
  masterKeyDialogHeader: createHeader('Master key generated'),
  masterKeyFieldLabel: 'Master Key:',
  incorrectPassword: 'Password was incorrect.',
  queries: 'Queries',
  queriesDialogTitle: (count: number) => `Queries (${count})`,
  newQueryDialogTitle: 'New Query Type',
  exportQueryForDwca: 'Export query for DwCA definition.',
  exportQueryForDwcaDialogTitle: 'Query XML for DwCA definition.',
  exportQueryForDwcaDialogHeader: createHeader(''),
  exportQueryAsReport: 'Define report based on query.',
  exportQueryAsLabel: 'Define label based on query.',
  newResourceTitle: (resourceName: string) => `New ${resourceName}`,
  labelName: 'Label Name',
  reportName: 'Report Name',
  createLabelDialogTitle: 'Labels',
  createLabelDialogHeader: createHeader('Create new label'),
  createReportDialogTitle: 'Reports',
  createReportDialogHeader: createHeader('Create new report'),
  recordSets: 'Record Sets',
  resources: 'Resources',
  appResources: 'App Resources',
  viewSets: 'View Sets',
  resourcesDialogTitle: 'Resources',
  resourcesDialogHeader: createHeader(
    'Choose the resource type you wish to edit:'
  ),
  repairTree: 'Repair Tree',
  trees: 'Trees',
  treesDialogTitle: 'Trees',
  manageUsers: 'Manage Users',
  manageUsersDialogTitle: 'Manage Users',
  workbench: 'WorkBench',
  chooseDwcaDialogTitle: 'Choose DwCA',
  dwcaDefinition: 'DwCA definition:',
  metadataResource: 'Metadata resource:',

  // Error Boundary
  errorBoundaryDialogTitle: 'Unexpected Error',
  errorBoundaryDialogHeader: createJsxHeader(
    'An unexpected error has occurred'
  ),
  errorBoundaryDialogMessage: (
    <>
      Please reload the page and try again. If this issue persists, please
      contact your IT support or if this is a Specify Cloud database, contact
      <a href="mailto:support@specifysoftware.org">
        support@specifysoftware.org
      </a>
      .
    </>
  ),
  backEndErrorDialogTitle: 'Server Error',
  backEndErrorDialogHeader: createHeader(
    'An error occurred communicating with the Specify 7 server.'
  ),
  backEndErrorDialogMessage: `
    Please reload the page and try again. If the issue persists, please
    contact your IT support, or if this is a Specify Cloud database, contact
    <a href="mailto:support@specifysoftware.org">
      support@specifysoftware.org
    </a>.`,

  // Search
  expressSearch: 'Express Search',
  primarySearch: 'Primary Search',
  secondarySearch: 'Secondary Search',
  running: 'Running...',
  noMatches: 'No Matches',

  // Unload Protection
  leavePageDialogTitle: 'Unsaved changes detected',
  leavePageDialogHeader: createHeader(
    'Are you sure you want to leave this page?'
  ),
  leave: 'Leave',

  // Notifications
  notificationsDialogTitle: 'Notifications',
  feedItemUpdated: 'Export feed item updated.',
  updateFeedFailed: 'Export feed update failed.',
  dwcaExportCompleted: 'DwCA export completed.',
  dwcaExportFailed: 'DwCA export failed.',
  queryExportToCsvCompleted: 'Query export to CSV completed.',
  queryExportToKmlCompleted: 'Query export to KML completed.',
  dataSetOwnershipTransferred: (userName: string, dataSetName: string) => `
    ${userName} transferred the ownership of the ${dataSetName} dataset
    to you.`,

  // OtherCollectionView
  noAccessToResource: `
    You do not have access to any collection containing this resource
    through the currently logged in account`,
  resourceInaccessible: `
    The requested resource cannot be accessed while logged into the
    current collection.`,
  selectCollection: 'Select one of the following collections:',
  loginToProceed: (collectionName: string) => `
    You can login to the collection, ${collectionName}, to proceed:`,

  // SpecifyApp
  versionMismatchDialogTitle: 'Version Mismatch',
  versionMismatchDialogHeader: createHeader(
    'Specify version does not match database version'
  ),
  versionMismatchDialogMessage: (
    specifySixVersion: string,
    databaseVersion: string
  ) => `
    The Specify version ${specifySixVersion} does not match the database
    version ${databaseVersion}.`,
  versionMismatchSecondDialogMessage:
    'Some features of Specify 7 may therefore fail to operate correctly.',
  resourceDeletedDialogTitle: 'Deleted',
  resourceDeletedDialogHeader: createHeader('Item deleted'),
  resourceDeletedDialogMessage: 'Item was deleted successfully.',
  appTitle: (baseTitle: string) => `${baseTitle} | Specify 7`,

  // StartApp
  sessionTimeOutDialogTitle: 'Access denied',
  sessionTimeOutDialogHeader: createHeader('Insufficient Privileges'),
  sessionTimeOutDialogMessage: `
    You lack sufficient privileges for that action, or your current
    session has been logged out.`,

  // UserTools
  logOut: 'Log out',
  changePassword: 'Change password',
  userToolsDialogTitle: 'User Tools',
});

export default commonText;
