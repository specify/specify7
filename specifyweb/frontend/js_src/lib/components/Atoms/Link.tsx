import React from 'react';
import type { LocalizedString } from 'typesafe-i18n';

import { commonText } from '../../localization/common';
import type { IR, RA, RR } from '../../utils/types';
import { className } from './className';
import type { IconProps } from './Icons';
import { icons } from './Icons';
import type { TagProps } from './wrapper';
import { wrap } from './wrapper';

/**
 * A wrapper for wrap() to generate links that have [href] attribute required
 */
// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
const linkComponent = <EXTRA_PROPS extends IR<unknown> = RR<never, never>>(
  name: string,
  className: string,
  initialProps?:
    | TagProps<'a'>
    | ((props: Readonly<EXTRA_PROPS> & TagProps<'a'>) => TagProps<'a'>)
) =>
  wrap<
    'a',
    EXTRA_PROPS & {
      readonly href: string;
      readonly children?:
        | JSX.Element
        | LocalizedString
        | RA<JSX.Element | LocalizedString | false | undefined>;
      readonly title?: LocalizedString | undefined;
      readonly 'aria-label'?: LocalizedString | undefined;
    }
  >(name, 'a', className, initialProps);

export const Link = {
  Default: linkComponent('Link.Default', className.link),
  NewTab: linkComponent('Link.NewTab', className.link, (props) => ({
    ...props,
    target: '_blank',
    rel: 'noopener',
    children: (
      <>
        {props.children}
        <span title={commonText.opensInNewTab()}>
          <span className="sr-only">{commonText.opensInNewTab()}</span>
          {icons.externalLink}
        </span>
      </>
    ),
  })),
  Small: linkComponent<{
    /*
     * A class name that is responsible for text and background color
     * Split into a separate prop in order to add a default value
     */
    readonly variant?: string;
  }>(
    'Link.Small',
    className.smallButton,
    ({
      variant = className.defaultSmallButtonVariant,
      className: classString = '',
      ...props
    }) => ({
      className: `${classString} ${variant}`,
      ...props,
    })
  ),
  Fancy: linkComponent(
    'Link.Fancy',
    `${className.niceButton} ${className.fancyButton}`
  ),
  Secondary: linkComponent(
    'Link.Secondary',
    `${className.niceButton} ${className.secondaryButton}`
  ),
  BorderedGray: linkComponent(
    'Link.BorderedGray',
    `${className.niceButton} ${className.borderedGrayButton}`
  ),
  Danger: linkComponent(
    'Link.Danger',
    `${className.niceButton} ${className.dangerButton}`
  ),
  Info: linkComponent(
    'Link.Info',
    `${className.niceButton} ${className.infoButton}`
  ),
  Warning: linkComponent(
    'Link.Warning',
    `${className.niceButton} ${className.warningButton}`
  ),
  Success: linkComponent(
    'Link.Success',
    `${className.niceButton} ${className.successButton}`
  ),
  Icon: linkComponent<IconProps>(
    'Link.Icon',
    `${className.icon} rounded`,
    ({ icon, ...props }) => ({
      ...props,
      'aria-label': props['aria-label'] ?? props.title,
      children: icons[icon],
    })
  ),
} as const;
