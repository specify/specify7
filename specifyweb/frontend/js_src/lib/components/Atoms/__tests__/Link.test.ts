import { commonText } from '../../../localization/common';
import { snapshot } from '../../../tests/reactUtils';
import { className } from '../className';
import { Link } from '../Link';

snapshot(Link.Default, { href: '#', children: commonText.close() });
snapshot(Link.NewTab, { href: '#', children: commonText.close() });
describe('Link.Small', () => {
  snapshot(Link.Small, { href: '#' }, 'default variant');
  snapshot(
    Link.Small,
    {
      href: '#',
      variant: className.infoButton,
      className: 'a',
    },
    'custom variant'
  );
});

snapshot(Link.Fancy, { href: '#', children: commonText.close() });
snapshot(Link.Secondary, { href: '#', children: commonText.close() });
snapshot(Link.BorderedGray, { href: '#', children: commonText.close() });
snapshot(Link.Danger, { href: '#', children: commonText.close() });
snapshot(Link.Info, { href: '#', children: commonText.close() });
snapshot(Link.Warning, { href: '#', children: commonText.close() });
snapshot(Link.Success, { href: '#', children: commonText.close() });
snapshot(Link.Icon, { href: '#', title: commonText.close(), icon: 'cog' });
