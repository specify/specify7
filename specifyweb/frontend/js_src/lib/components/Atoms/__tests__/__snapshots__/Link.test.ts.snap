// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link.BorderedGray renders without errors 1`] = `
<DocumentFragment>
  <a
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:dark:ring-neutral-500 disabled:ring-gray-400 disabled:text-gray-500 
    dark:disabled:!bg-neutral-700 gap-2 inline-flex items-center capitalize justify-center shadow-sm button hover:brightness-90 dark:hover:brightness-125 bg-[color:var(--secondary-button-color)] text-gray-800 shadow-sm
dark:text-gray-100 ring-1 ring-gray-400 shadow-sm dark:ring-0
    disabled:ring-gray-400 disabled:dark:ring-neutral-500"
    href="#"
  >
    Close
  </a>
</DocumentFragment>
`;

exports[`Link.Danger renders without errors 1`] = `
<DocumentFragment>
  <a
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:dark:ring-neutral-500 disabled:ring-gray-400 disabled:text-gray-500 
    dark:disabled:!bg-neutral-700 gap-2 inline-flex items-center capitalize justify-center shadow-sm dialog-icon-error hover:brightness-90 dark:hover:brightness-150 bg-[color:var(--danger-button-color)] text-white"
    href="#"
  >
    Close
  </a>
</DocumentFragment>
`;

exports[`Link.Default renders without errors 1`] = `
<DocumentFragment>
  <a
    class="link"
    href="#"
  >
    Close
  </a>
</DocumentFragment>
`;

exports[`Link.Fancy renders without errors 1`] = `
<DocumentFragment>
  <a
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:dark:ring-neutral-500 disabled:ring-gray-400 disabled:text-gray-500 
    dark:disabled:!bg-neutral-700 gap-2 inline-flex items-center capitalize justify-center shadow-sm bg-gray-300 hover:bg-brand-200 dark:bg-neutral-600
    hover:dark:bg-brand:400 text-gray-800 dark:text-white text-center"
    href="#"
  >
    Close
  </a>
</DocumentFragment>
`;

exports[`Link.Icon renders without errors 1`] = `
<DocumentFragment>
  <a
    aria-label="Close"
    class="icon link rounded"
    href="#"
    title="Close"
  >
    <svg
      aria-hidden="true"
      class="w-6 h-6 flex-shrink-0"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M11.49 3.17c-.38-1.56-2.6-1.56-2.98 0a1.532 1.532 0 01-2.286.948c-1.372-.836-2.942.734-2.106 2.106.54.886.061 2.042-.947 2.287-1.561.379-1.561 2.6 0 2.978a1.532 1.532 0 01.947 2.287c-.836 1.372.734 2.942 2.106 2.106a1.532 1.532 0 012.287.947c.379 1.561 2.6 1.561 2.978 0a1.533 1.533 0 012.287-.947c1.372.836 2.942-.734 2.106-2.106a1.533 1.533 0 01.947-2.287c1.561-.379 1.561-2.6 0-2.978a1.532 1.532 0 01-.947-2.287c.836-1.372-.734-2.942-2.106-2.106a1.532 1.532 0 01-2.287-.947zM10 13a3 3 0 100-6 3 3 0 000 6z"
        fill-rule="evenodd"
      />
    </svg>
  </a>
</DocumentFragment>
`;

exports[`Link.Info renders without errors 1`] = `
<DocumentFragment>
  <a
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:dark:ring-neutral-500 disabled:ring-gray-400 disabled:text-gray-500 
    dark:disabled:!bg-neutral-700 gap-2 inline-flex items-center capitalize justify-center shadow-sm dialog-icon-info hover:brightness-90 dark:hover:brightness-150 bg-[color:var(--info-button-color)] text-white"
    href="#"
  >
    Close
  </a>
</DocumentFragment>
`;

exports[`Link.NewTab renders without errors 1`] = `
<DocumentFragment>
  <a
    class="link"
    href="#"
    rel="noopener"
    target="_blank"
  >
    Close
    <span
      title="(opens in a new tab)"
    >
      <span
        class="sr-only"
      >
        (opens in a new tab)
      </span>
      <svg
        aria-hidden="true"
        class="w-6 h-6 flex-shrink-0"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 3a1 1 0 100 2h2.586l-6.293 6.293a1 1 0 101.414 1.414L15 6.414V9a1 1 0 102 0V4a1 1 0 00-1-1h-5z"
        />
        <path
          d="M5 5a2 2 0 00-2 2v8a2 2 0 002 2h8a2 2 0 002-2v-3a1 1 0 10-2 0v3H5V7h3a1 1 0 000-2H5z"
        />
      </svg>
    </span>
  </a>
</DocumentFragment>
`;

exports[`Link.Secondary renders without errors 1`] = `
<DocumentFragment>
  <a
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:dark:ring-neutral-500 disabled:ring-gray-400 disabled:text-gray-500 
    dark:disabled:!bg-neutral-700 gap-2 inline-flex items-center capitalize justify-center shadow-sm button hover:brightness-90 dark:hover:brightness-125 bg-[color:var(--secondary-button-color)] text-gray-800 shadow-sm
dark:text-gray-100"
    href="#"
  >
    Close
  </a>
</DocumentFragment>
`;

exports[`Link.Small custom variant 1`] = `
<DocumentFragment>
  <a
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:dark:ring-neutral-500 disabled:ring-gray-400 disabled:text-gray-500 
    dark:disabled:!bg-neutral-700 gap-2 inline-flex items-center capitalize justify-center shadow-sm !py-1 !px-2 a dialog-icon-info hover:brightness-90 dark:hover:brightness-150 bg-[color:var(--info-button-color)] text-white"
    href="#"
  />
</DocumentFragment>
`;

exports[`Link.Small default variant 1`] = `
<DocumentFragment>
  <a
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:dark:ring-neutral-500 disabled:ring-gray-400 disabled:text-gray-500 
    dark:disabled:!bg-neutral-700 gap-2 inline-flex items-center capitalize justify-center shadow-sm !py-1 !px-2 button hover:brightness-90 dark:hover:brightness-125 bg-[color:var(--secondary-button-color)] text-gray-800 shadow-sm
dark:text-gray-100 ring-1 ring-gray-400 shadow-sm dark:ring-0
    disabled:ring-gray-400 disabled:dark:ring-neutral-500"
    href="#"
  />
</DocumentFragment>
`;

exports[`Link.Success renders without errors 1`] = `
<DocumentFragment>
  <a
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:dark:ring-neutral-500 disabled:ring-gray-400 disabled:text-gray-500 
    dark:disabled:!bg-neutral-700 gap-2 inline-flex items-center capitalize justify-center shadow-sm dialog-icon-success hover:brightness-90 dark:hover:brightness-150 bg-[color:var(--success-button-color)] text-white"
    href="#"
  >
    Close
  </a>
</DocumentFragment>
`;

exports[`Link.Warning renders without errors 1`] = `
<DocumentFragment>
  <a
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:dark:ring-neutral-500 disabled:ring-gray-400 disabled:text-gray-500 
    dark:disabled:!bg-neutral-700 gap-2 inline-flex items-center capitalize justify-center shadow-sm dialog-icon-warning hover:brightness-90 dark:hover:brightness-150 bg-[color:var(--warning-button-color)] text-white"
    href="#"
  >
    Close
  </a>
</DocumentFragment>
`;
